app/uk/gov/hmrc/agentservicesaccount/forms/UpdateMoneyLaunderingSupervisionForm.scala

```diff
@@ -30,39 +30,103 @@ object UpdateMoneyLaunderingSupervisionForm {
   private val supervisoryNumberRegex = """^[A-Za-z0-9\,\.\'\-\/\ ]{0,200}$""".r // remove all spaces from input before matching to ensure correct digit count
   private val trimmedText = text.transform[String](x => x.trim, x => x)
 
-  private val invalidDateConstraint: Constraint[(String, String, String)] = Constraint[(String, String, String)] { data: (String, String, String) =>
-    val (day, month, year)  = data
-
+  private def invalidDateCheck(day: String, month: String, year: String): Either[(String, String), String] = {
     Try {
       require(year.length == 4, "Year must be 4 digits")
```

Comment:
Should this error message be a key in the messages file?

Reply:
No, it's just a message assigned to an error caught by the `Try... Catch` that is interpreted to a user message using the key.

Reply:
Do we need this check on the length of the year? Does the attempt to convert to a localdate meet the required validation?

Reply:
I had thought about it as I was reviewing the logic but I believe the error message is the same. If the error message were different it'd need changing.